app/v1/controllers/validators/AmendOtherReliefsRulesValidator.scala

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v1.controllers.validators
+
+import api.controllers.validators.RulesValidator
+import api.controllers.validators.resolvers.{ResolveIsoDate, ResolveParsedNumber}
+import api.models.errors._
+import cats.data.Validated
+import cats.data.Validated.Invalid
+import cats.implicits.toFoldableOps
+import v1.models.request.amendOtherReliefs._
+
+object AmendOtherReliefsRulesValidator extends RulesValidator[AmendOtherReliefsRequestData] {
+
+  private val resolveParsedNumber = ResolveParsedNumber()
+
+  private val stringRegex = "^[0-9a-zA-Z{À-˿’}\\- _&`():.'^]{1,90}$".r
+
+  private def zipAndValidate[FIELD](maybeFields: Option[Seq[FIELD]],
```

Comment:
could these private defs go below the public def?  Good to always have the "public interface" at the top of any class or object.